File: backend/service-community/models/community.go
package models

import "time"

type Community struct {
	CommunityId   string `gorm:"primaryKey"`
	CommunityName string
	CreatorId     string
	Description   string `gorm:"type:text"`
	Rules         string `gorm:"type:text"`
	IconImage     string
	BannerImage   string
	IsPending     bool
	CreatedAt     time.Time
}

type CommunityCategory struct {
	Category string `gorm:"unique"`
}

type CommunityCategoryRelation struct {
	CommunityId string `gorm:"primaryKey"`
	Category    string `gorm:"primaryKey"`
}

type CommunityMember struct {
	CommunityId string `gorm:"primaryKey"`
	UserId      string `gorm:"primaryKey"`
	Role        string
	JoinedAt    time.Time
}

type CommunityJoinRequest struct {
	RequestId   string `gorm:"primaryKey"`
	CommunityId string
	UserId      string
	Status      string
	RequestedAt time.Time
}

File: frontend/src/types/community.ts
export interface Community {
    community_id: string;
    community_name: string;
    creator_id: string;
    categories: string[];
    description: string;
    rules: string;
    created_at: string;
    icon_image: string;
    banner_image: string;
    member_count: number;
    role: string;
    is_pending: boolean;
}

export interface GetCommunitiesResponse {
    communities: Community[];
}

export interface GetCategoriesResponse {
    categories: string[];
}

export interface CommunityMember {
    user_id: string;
    role: string;
}

export interface GetCommunityMembersResponse {
    members: CommunityMember[];
}
